frontend/components/App.tsx

```diff
@@ -109,41 +101,44 @@ function App() {
     return () => {
       ipcRenderer.removeListener('db-lists', dbListFromBackend);
     };
-  });
+  }, [selectedTable, dbTables]);
+
 
   // Handle async calls
   const asyncCount = useRef(0);
+  // 
   useEffect(() => {
-    const { issued, resolved, asyncList } = menuState.loading;
-    if (issued - resolved > asyncCount.current) {
-      submitAsyncToBackend(issued, asyncList, invoke, menuDispatch);
+    const { issued, asyncList } = menuState.loading;
+    // Check if there are pending async tasks
+    if (issued > 0 && issued > asyncCount.current) {
+      // Dispatch the submitAsyncToBackend thunk action
+      dispatch(submitAsyncToBackend({ issued, asyncList, invoke }));
     }
-    asyncCount.current = issued - resolved;
-  }, [menuState.loading, menuDispatch]);
+    // Update the asyncCount ref with the latest issued count
+    asyncCount.current = issued;
+  }, [menuState.loading, dispatch]);
 
+  
   // Populate initial dblist
   useEffect(() => {
     const dbListFromBackend = (dbLists: DbListsInterface) => {
       setDBInfo(dbLists.databaseList);
       setTables(dbLists.tableList);
-      appViewDispatch({
-        type: 'IS_PG_CONNECTED',
-        payload: dbLists.databaseConnected.PG,
-      });
-      appViewDispatch({
-        type: 'IS_MYSQL_CONNECTED',
-        payload: dbLists.databaseConnected.MySQL,
-      });
+      dispatch({ type: 'appView/setPGConnected', payload: dbLists.databaseConnected.PG });
+      dispatch({ type: 'appView/setMYSQLConnected', payload: dbLists.databaseConnected.MySQL });
     };
-    menuDispatch({
-      type: 'ASYNC_TRIGGER',
-      loading: 'LOADING',
-      options: {
-        event: 'return-db-list',
-        callback: dbListFromBackend,
+    dispatch({
```

Comment:
Replace current Context API code with correct redux function calls. You will need to import those functions from the slices